src/catalog.rs

```diff
@@ -223,310 +393,271 @@ pub trait TableCatalog: Sync + Send {
-            }),
-        )
-    }
 }
 
 #[async_trait]
-impl TableCatalog for DefaultCatalog {
-    async fn load_database(&self, database_id: DatabaseId) -> Result<SeafowlDatabase> {
-        let all_columns = self
-            .repository
-            .get_all_columns_in_database(database_id)
-            .await
-            .map_err(Self::to_sqlx_error)?;
-
-        // NB we can't distinguish between a database without tables and a database
-        // that doesn't exist at all due to our query.
-
-        // Turn the list of all collections, tables and their columns into a nested map.
-
-        let collections: HashMap<Arc<str>, Arc<SeafowlCollection>> = all_columns
-            .iter()
-            .group_by(|col| &col.collection_name)
-            .into_iter()
-            .map(|(cn, cc)| self.build_collection(cn, cc))
-            .collect();
-
-        // TODO load the database name too
-        let name: Arc<str> = Arc::from(DEFAULT_DB);
-
-        Ok(SeafowlDatabase {
-            name: name.clone(),
-            collections,
-            staging_schema: self.staging_schema.clone(),
-            system_schema: Arc::new(SystemSchemaProvider::new(
-                name,
-                Arc::new(self.clone()),
-            )),
-        })
-    }
-
-    async fn load_database_ids(&self) -> Result<HashMap<String, DatabaseId>> {
-        let all_db_ids = self
-            .repository
-            .get_all_database_ids()
-            .await
-            .map_err(Self::to_sqlx_error)?;
-
-        Ok(HashMap::from_iter(all_db_ids))
-    }
 
-    async fn create_table(
-        &self,
-        collection_id: CollectionId,
-        table_name: &str,
-        schema: &Schema,
-        uuid: Uuid,
-    ) -> Result<(TableId, TableVersionId)> {
+impl CatalogStore for RepositoryStore {
+    async fn create(&self, name: &str) -> Result<DatabaseId> {
         self.repository
-            .create_table(collection_id, table_name, schema, uuid)
+            .create_database(name)
             .await
             .map_err(|e| match e {
                 RepositoryError::UniqueConstraintViolation(_) => {
-                    Error::TableAlreadyExists {
-                        name: table_name.to_string(),
+                    Error::CatalogAlreadyExists {
+                        name: name.to_string(),
                     }
                 }
-                RepositoryError::FKConstraintViolation(_) => {
-                    Error::CollectionDoesNotExist { id: collection_id }
-                }
-                RepositoryError::SqlxError(e) => Error::SqlxError(e),
+                e => e.into(),
             })
     }
 
-    async fn delete_old_table_versions(&self, table_id: TableId) -> Result<u64, Error> {
+    async fn get(&self, name: &str) -> Result<DatabaseRecord> {
+        match self.repository.get_database(name).await {
+            Ok(database) => Ok(database),
+            Err(RepositoryError::SqlxError(sqlx::error::Error::RowNotFound)) => {
```

Comment:
minor: could map_err here?

Reply:
Yup that looks better (changed it it in `SchemaStore::get` and `TableStore::get` as well).